src/ServerMain.cpp

```diff
@@ -122,14 +126,20 @@ int main(int argc, char** argv) {
     return EXIT_FAILURE;
   }
 
+  // If no vocabulary basename is given, use the index basename.
+  if (baseNameVocabulary.empty()) {
+    baseNameVocabulary = baseNameIndex;
+  }
+
   LOG(INFO) << EMPH_ON << "QLever Server, compiled on "
             << qlever::version::DatetimeOfCompilation << " using git hash "
             << qlever::version::GitShortHash() << EMPH_OFF << std::endl;
 
   try {
     Server server(port, numSimultaneousQueries, memoryMaxSize,
                   std::move(accessToken), !noPatternTrick);
-    server.run(indexBasename, text, !noPatterns, !onlyPsoAndPosPermutations);
+    server.run(baseNameIndex, baseNameVocabulary, text, !noPatterns,
+               !onlyPsoAndPosPermutations);
```

Comment:
I think one thing you could do (the outer interface that interacts with the control script is mostly you area)
is to set up a simple struct `ServerConfig` that is then passed to the constructor as well as the `run()` function where they grap their respectively needed arguments. That makes it much easier to add additional arguments.
(Probably we need a similar struct `IndexConfig` that then becomes part of the server config for exactly the same reason).
Then it will become much easier to add additional arguments.
Are you interested in setting this up as a separate PR, or should I do this?